configuration/pih/liquibase/liquibase-mexico.xml

```diff
@@ -13,6 +13,66 @@
         </preConditions>            
         <sqlFile endDelimiter=";" path="sql/migrate_adjustment_disorder_diagnosis.sql" relativeToChangelogFile="true" stripComments="true" />
     </changeSet>
+    <changeSet id="20230208-void-duplicate-empty-encounters" author="dmdesimone">
+	    <preConditions onFail="MARK_RAN">
+			<not><sqlCheck expectedResult="0">
+					select  count(*) 
+					from encounter e
+					where e.voided = 0 
+					and not EXISTS 
+						(select 1 from obs o
+						where o.encounter_id = e.encounter_id 
+						and o.voided = 0)
+					and EXISTS -- at least one obs referred to as a previous obs in other encounter 
+						(select 1 from obs o4
+						inner join obs o5 on o5.previous_version = o4.obs_id and o5.encounter_id != o4.encounter_id 
+						where o4.encounter_id = e.encounter_id)
+					and not EXISTS -- there are no obs in the encounter that are NOT referred as a previous obs in another encounter   
```

Comment:
I'm probably missing something obvious, but are both these clauses needed... ie, don't we just need to check that there are no obs in the encounter that are NOT referred as a previous obs in another encounter.

Reply:
Yeah, I went back and forth on this.  If you check that there is at least one obs that IS referred as a previous obs, you are guaranteeing that you get this specific case.  You could argue that that we should also void non-voided encounters that are completely empty, but that's a different case.

But yeah - I can remove that first not exists clause and will do that.

Reply:
But yeah, let me know if this makes sense @mogoodrich before I merge this in.

Reply:
Actually, @dmdesimone  I think the test for voided obs is a good one to have (looks like you removed it).

It was middle case that was confusing me.... if we confirm that there are no obs in the encounter that are NOT referred as a previous obs in another encounter, doesn't that guarantee that there's at least one obs in the encounter that is *not* referred to as a previous obs in another encoounter.  ie, I was suggesting getting rid of the middle clause:

```
	and EXISTS -- at least one obs referred to as a previous obs in other encounter 
						(select 1 from obs o4
						inner join obs o5 on o5.previous_version = o4.obs_id and o5.encounter_id != o4.encounter_id 
						where o4.encounter_id = e.encounter_id)
```

... not the one that checks for voided....

Reply:
Yeah, so this is a bit confusing but... the third clause (there are no obs in the encounter that are NOT referred as a previous obs in another encounter ) includes the case of encounters completely empty of non-voided obs.  So that's why I removed that first unnecessary clause and also why that second clause is needed.  Again, we could make an argument that we could void encounters that are completely empty of obs but some of those will be a different case than the one we are correcting for.

So just to test this, when I removed the first clause and ran counts on the copy of production I got the same # of results as when the first clause was there.  In contrary - when I remove the second clause, I got many more results because again, it will include a bunch of empty encounters unrelated to this issue.

Reply:
Got it, make sense.  I'm forgetting, did this only happen with a single encounter type, and therefore did we want to restrict by encounter type to to be say...